File: src/main/kotlin/gbl/tag/type/application/GblApplication.kt
package gbl.tag.type.application

import gbl.tag.GblType
import gbl.tag.TagHeader
import gbl.tag.Tag
import gbl.tag.TagWithHeader

data class GblApplication(
    override val tagHeader: TagHeader,
    override val tagType: GblType,
    val applicationData: ApplicationData,
    override val tagData: ByteArray
) : Tag, TagWithHeader {
    override fun copy(): Tag {
        return GblApplication(
            tagHeader = tagHeader,
            tagType = tagType,
            applicationData = applicationData,
            tagData = arrayOf<Byte>().toByteArray()
        )
    }

    companion object {
        const val APP_LENGTH = 13
    }
}

File: python-library/tag/type/application/gbl_application.py
from dataclasses import dataclass
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from ...gbl_type import GblType
    from ...tag_header import TagHeader
    from ...tag import Tag

from .application_data import ApplicationData


@dataclass
class GblApplication:
    """GBL Application tag"""

    tag_header: 'TagHeader'
    application_data: ApplicationData
    tag_data: bytes

    # Constant як у Kotlin companion object
    APP_LENGTH: int = 13

    @property
    def tag_type(self) -> 'GblType':
        """Get tag type"""
        from ...gbl_type import GblType
        return GblType.APPLICATION

    def copy(self) -> 'Tag':
        """Create a copy of the tag"""
        return GblApplication(
            tag_header=self.tag_header,
            application_data=self.application_data,
            tag_data=bytes()  # Empty byte array як у Kotlin
        )